examples/vault-prod-ready/main.tf

```diff
@@ -0,0 +1,534 @@
+  connection_draining         = true
+  connection_draining_timeout = 60
+
+  # Perform TCP health checks on Consul's client port.
+  health_check {
+    target              = "TCP:8200"
+    interval            = 30
+    timeout             = 10
+    healthy_threshold   = 2
+    unhealthy_threshold = 2
+  }
+
+  # The ELB can be used to reach the management interface
+  listener {
+    instance_port     = 8200
+    instance_protocol = "http"
+    lb_port           = 8200
+    lb_protocol       = "http"
+  }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE A SECURITY GROUP THAT CONTROLS WHAT TRAFFIC CAN GO IN AND OUT OF THE ELB OF VAULT
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "aws_security_group" "elb_vault" {
+  name   = var.vault_cluster_name
+  vpc_id = var.vpc_id
+}
+
+resource "aws_security_group_rule" "allow_all_outbound_vault" {
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.elb_vault.id
+}
+
+resource "aws_security_group_rule" "allow_vault_inbound_from_elb" {
+  type = "ingress"
+  from_port = 8200
+  to_port = 8200
+  protocol = "tcp"
+  security_group_id = aws_security_group.elb_vault.id
+  source_security_group_id = module.vault_cluster.security_group_id
+}
+
+resource "aws_security_group_rule" "allow_openvpntovault_inbound_from_elb" {
+  type = "ingress"
+  from_port = 8200
+  to_port = 8200
+  protocol = "tcp"
+  security_group_id = aws_security_group.elb_vault.id
+  source_security_group_id = var.security_group_id
+}
+## Allow clients to connect from within the provided security group
+resource "aws_security_group_rule" "allow_vault_inbound_from_subnets" {
+  type = "ingress"
+  from_port = 8200
+  to_port = 8200
+  protocol = "tcp"
+  security_group_id = aws_security_group.elb_vault.id
+  cidr_blocks = var.vpc_cidr_blocks
+}
+#Assign ASG to ELB
+resource "aws_autoscaling_attachment" "elb_vault" {
+  autoscaling_group_name = module.vault_cluster.asg_name
+  elb                    = var.vault_cluster_name
+}
+
+# Secret that holds the Consul master token
+resource "aws_secretsmanager_secret" "consul_token" {
+  name_prefix = "${var.consul_cluster_name}-token"
+}
+
+# Random uuid used as master token
+resource "random_uuid" "consul_token" {}
+
+# Secret version updated with the random uuid
+resource "aws_secretsmanager_secret_version" "consul_token" {
+  secret_id     = aws_secretsmanager_secret.consul_token.id
+  secret_string = random_uuid.consul_token.result
+}
+
+# Secret that holds the gossip encryption key
+resource "aws_secretsmanager_secret" "gossip_encryption_key" {
+  name_prefix = "gossip_encryption_key"
+}
+
+# Random uuid used as gossip encryption key
+resource "random_string" "gossip_encryption_key" {
```

Comment:
This too ends up stored in state. Again, would be better to add it to secrets manager out of band.

Reply:
Agree